chacra/async/recurring.py

```diff
@@ -84,21 +84,27 @@ def purge_repos(_now=None):
         project_repos = models.Repo.filter_by(project=p)
 
         # get the configuration for current project
-        ref_names = purge_rotation[project_name].keys()
         purge_project = purge_rotation.get(project_name)
-
-        for ref_name in ref_names:
-            purge_ref = purge_project[ref_name]
-            lifespan = now - datetime.timedelta(days=purge_ref.get('days', 14))
-            keep_minimum = purge_ref.get('keep_minimum', 0)
-            ref_repos = project_repos.filter_by(ref=ref_name)
-
-            # filter the query further, with the offset being the
-            # minimum to keep, and lifespan how old they should be
-            repos = ref_repos.filter(
-                    models.Repo.modified < lifespan).order_by(
-                        desc(models.Repo.modified)).offset(keep_minimum).all()
-            delete_repositories(repos, lifespan, keep_minimum)
+        project_filters = purge_project.keys()
+        # can be ref and/or flavor
+        for proj_filter in project_filters:
```

Comment:
you can do:

    for proj_filter, repo_attrs in purge_project.items():

Because `items()` will get you both the key (`proj_filter`) and the value of that key (`repo_attrs`)